From 414ee790ea81a3be365d3ee7cee3b6ccc9bb8cd4 Mon Sep 17 00:00:00 2001
From: OpenWrt community <openwrt-devel@lists.openwrt.org>
Date: Thu, 16 Mar 2023 16:19:09 +0530
Subject: [PATCH 038/500] OpenWrt: 482-mtd-spi-nor-add-gd25q512.patch

spi-nor/gigadevic: add gd25q512

Change-Id: I88bac60ead84f5dccae5d42e0a808f5f5dd825de
Signed-off-by: Ram Chandra Jangir <quic_rjangir@quicinc.com>
(cherry picked from commit dd1d089e140aa5aaf4b0a2933b3562541aab0d45)
---
 drivers/mtd/spi-nor/gigadevice.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/mtd/spi-nor/gigadevice.c b/drivers/mtd/spi-nor/gigadevice.c
index d0c5e3ac3c88..2be2bda6257e 100644
--- a/drivers/mtd/spi-nor/gigadevice.c
+++ b/drivers/mtd/spi-nor/gigadevice.c
@@ -71,6 +71,11 @@ static const struct flash_info gigadevice_nor_parts[] = {
 		FLAGS(SPI_NOR_HAS_LOCK | SPI_NOR_HAS_TB | SPI_NOR_TB_SR_BIT6)
 		FIXUP_FLAGS(SPI_NOR_4B_OPCODES)
 		.fixups = &gd25q256_fixups },
+	{ "gd25q512", INFO(0xc84020, 0, 64 * 1024, 1024)
+		FLAGS(SPI_NOR_HAS_LOCK | SPI_NOR_HAS_TB)
+		FIXUP_FLAGS(SPI_NOR_4B_OPCODES)
+		NO_SFDP_FLAGS(SECT_4K | SPI_NOR_DUAL_READ |
+			      SPI_NOR_QUAD_READ) },
 };
 
 const struct spi_nor_manufacturer spi_nor_gigadevice = {
-- 
2.34.1

